docs/Toy/toy/dialects/toy.py

```diff
@@ -121,11 +111,11 @@ def get_data(self) -> list[float]:
         return list(self.value.get_values())
 
 
-class InferAddOpShapeTrait(ToyShapeInferenceTrait):
-    @classmethod
-    def infer_shape(cls, op: Operation) -> None:
+class InferAddOpShapeInferencePattern(RewritePattern):
+    def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter, /):
```

Comment:
Is it worth using the decorator here, or is the idea to keep the toy files as simple as possible?